destination/destination.go

```diff
@@ -0,0 +1,112 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/conduitio-labs/conduit-connector-clickhouse/config"
+	"github.com/conduitio-labs/conduit-connector-clickhouse/destination/writer"
+	"github.com/conduitio-labs/conduit-connector-clickhouse/repository"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+// Writer defines a writer interface needed for the Destination.
+type Writer interface {
+	Write(context.Context, sdk.Record) error
+}
+
+// A Destination represents the destination connector.
+type Destination struct {
+	sdk.UnimplementedDestination
+
+	repo   *repository.ClickHouse
+	writer Writer
+	cfg    config.General
+}
+
+// NewDestination initialises a new Destination.
+func NewDestination() sdk.Destination {
+	return sdk.DestinationWithMiddleware(&Destination{}, sdk.DefaultDestinationMiddleware()...)
+}
+
+// Parameters returns a map of named Parameters that describe how to configure the Source.
+func (d *Destination) Parameters() map[string]sdk.Parameter {
+	return map[string]sdk.Parameter{
+		config.URL: {
+			Default:     "",
+			Required:    true,
+			Description: "The connection string to connect to ClickHouse database.",
+		},
+		config.Table: {
+			Default:     "",
+			Required:    true,
+			Description: "The table name of the table in ClickHouse that the connector should write to, by default.",
+		},
+		config.PrimaryColumns: {
+			Default:  "",
+			Required: false,
+			Description: "The list of columns, separated by commas, " +
+				"with constraints that uniquely identifies each record.",
+		},
+	}
+}
+
+// Configure parses and stores configurations, returns an error in case of invalid configuration.
+func (d *Destination) Configure(_ context.Context, cfg map[string]string) (err error) {
+	d.cfg, err = config.Parse(cfg)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Open initializes a publisher client.
+func (d *Destination) Open(ctx context.Context) (err error) {
+	d.repo, err = repository.New(d.cfg.URL)
+	if err != nil {
+		return fmt.Errorf("new repository: %w", err)
+	}
+
+	d.writer, err = writer.New(ctx, writer.Params{
+		Repo:           d.repo,
+		Table:          d.cfg.Table,
+		PrimaryColumns: d.cfg.PrimaryColumns,
+	})
+	if err != nil {
+		return fmt.Errorf("new writer: %w", err)
+	}
+
+	return nil
+}
+
+// Write writes records into a Destination.
+func (d *Destination) Write(ctx context.Context, records []sdk.Record) (int, error) {
+	for i, r := range records {
+		err := d.writer.Write(ctx, r)
```

Comment:
I think that the length of the line allows us to replace this with the following: `if err := d.writer.Write(ctx, r); err != nil {`.